components/helpers/ajax.js

```diff
@@ -0,0 +1,108 @@
+// generic ajax request function
```

Comment:
worth commenting that this needs refactoring into dependency rather than window.ajax?